components/script/dom/serviceworkerglobalscope.rs

```diff
@@ -185,61 +149,44 @@ impl ServiceWorkerGlobalScope {
-                timer_tx.send((worker_for_route.clone(), event)).unwrap();
-            });
+            // TODO XXXcreativcoder use this timer_ipc_port, when we have a service worker instance here
+            let (timer_ipc_chan, _timer_ipc_port) = ipc::channel().unwrap();
+            let (timer_chan, timer_port) = channel();
 
             let global = ServiceWorkerGlobalScope::new(
-                init, url, id, devtools_mpsc_port, runtime,
-                parent_sender.clone(), own_sender, receiver,
-                timer_ipc_chan, timer_rx, client);
-            // FIXME(njn): workers currently don't have a unique ID suitable for using in reporter
-            // registration (#6631), so we instead use a random number and cross our fingers.
+                init, url, pipeline_id, devtools_mpsc_port, runtime,
+                own_sender, receiver,
+                timer_ipc_chan, timer_port, swmanager_sender, scope_url);
             let scope = global.upcast::<WorkerGlobalScope>();
 
             unsafe {
                 // Handle interrupt requests
                 JS_SetInterruptCallback(scope.runtime(), Some(interrupt_callback));
             }
 
-            if scope.is_closing() {
-                return;
-            }
+            scope.execute_script(DOMString::from(source));
 
-            {
-                let _ar = AutoWorkerReset::new(global.r(), worker);
-                scope.execute_script(DOMString::from(source));
-            }
-
-
-            let reporter_name = format!("service-worker-reporter-{}", random::<u64>());
-            scope.mem_profiler_chan().run_with_memory_reporting(|| {
-                // Service workers are time limited
-                let sw_lifetime = Instant::now();
+            // Service workers are time limited
+            spawn_named("SWTimeoutThread".to_owned(), move || {
                 let sw_lifetime_timeout = PREFS.get("dom.serviceworker.timeout_seconds").as_u64().unwrap();
-                while let Ok(event) = global.receive_event() {
-                    if scope.is_closing() {
-                        break;
-                    }
-                    global.handle_event(event);
-                    if sw_lifetime.elapsed().as_secs() == sw_lifetime_timeout {
-                        break;
-                    }
+                thread::sleep(Duration::new(sw_lifetime_timeout, 0));
+                let _ = timer_chan.send(());
+            });
+
+            // TODO XXXcreativcoder bring back run_with_memory_reporting when things are more concrete here.
+            while let Ok(event) = global.receive_event() {
+                if !global.handle_event(event) {
+                    break;
                 }
-            }, reporter_name, parent_sender, CommonScriptMsg::CollectReports);
+            }
         });
     }
 
-    fn handle_event(&self, event: MixedMessage) {
+    fn handle_event(&self, event: MixedMessage) -> bool {
```

Comment:
Okay, so is this what was intended by the previous comment ? Other events return `true`,and only for timeout event i am returning a `false`, which halts the service worker.